proposals/lip-0058.md

```diff
@@ -736,21 +806,23 @@ Before the transactions of a block `b` are executed, the following logic is exec
 
 ### Block Creation
```

Comment:
Also here, it is not clear to me when would this logic be called.

Reply:
I improved this section and reference the concrete block generation stage defined in LIP 55. Note that as far as I understand both the header initialization and header finalization stage are in the engine domain.

Reply:
I think we can still improve by referencing exactly in which stage this happens.

Reply:
same problem, now it's fine, resolving.